src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -249,26 +249,26 @@ public TagHelperContent EndTagHelperWritingScope()
                 _originalWriter = null;
             }
 
-            var tagHelperContentWrapperTextWriter = new TagHelperContentWrapperTextWriter(Output.Encoding);
+            var tagHelperContent = new DefaultTagHelperContent();
             var razorWriter = writer as RazorTextWriter;
             if (razorWriter != null)
             {
-                razorWriter.CopyTo(tagHelperContentWrapperTextWriter);
+                tagHelperContent.Append(razorWriter.Buffer);
             }
             else
             {
                 var stringCollectionTextWriter = writer as StringCollectionTextWriter;
                 if (stringCollectionTextWriter != null)
                 {
-                    stringCollectionTextWriter.CopyTo(tagHelperContentWrapperTextWriter, HtmlEncoder);
+                    tagHelperContent.Append(stringCollectionTextWriter.Content);
                 }
                 else
                 {
-                    tagHelperContentWrapperTextWriter.Write(writer.ToString());
+                    tagHelperContent.Append(writer.ToString());
```

Comment:
And do we have tests demonstrating the `StringWriter` case doesn't lead to double-encoding?